common/tzif.ksy

```diff
@@ -0,0 +1,279 @@
+        type: tz_leap_second_record(is_ts64)
+        repeat: expr
+        repeat-expr: _parent.header.num_leap_second_records
+        doc: |
+          Array of leap-second records sorted in ascending order of occurrence
+          time for successive leap-second corrections. The first leap-second
+          record, if present, MUST have a non-negative occurrence time and MUST
+          have a correction value of either negative one (-1) or one (1). Each
+          succeeding leap-second record MUST have an occurrence time at least
+          2419199 greater than that of the previous record. The correction
+          values of two successive leap-second records MUST differ by exactly
+          one (1).
+      - id: is_std_flags
+        # standard/wall indicators
+        type: u1
+        repeat: expr
+        repeat-expr: _parent.header.num_is_std_flags
+        doc: |
+          Array of values indicating whether the transition times associated
+          with the corresponding local time types were specified as standard
+          time or wall-clock time. If the length of the array is zero then all
+          transition times associated with local time types are assumed to be
+          specified as wall-clock time. Each value MUST be 0 or 1. A value of
+          one (1) indicates standard time. A value of zero (0) indicates
+          wall-clock time. The value MUST be one (1) if the corresponding
+          UT/local indicator in the `is_ut_flags` array is one (1).
+      - id: is_ut_flags
+        # UT/local indicators
+        type: u1
+        repeat: expr
+        repeat-expr: _parent.header.num_is_ut_flags
+        doc: |
+          Array of values indicating whether the transition times associated
+          with the corresponding local time types were specified as UT or local
+          time. If the length of the array is zero then all transition times
+          associated with local time types are assumed to be specified as local
+          time. Each values MUST be 0 or 1. A value of one (1) indicates UT. A
+          value of zero (0) indicates local time. If the value is one (1) then
+          the corresponding standard/wall indicator in `is_std_flags` MUST also
+          be set to one (1).
+  tz_local_time_type:
+    doc-ref: https://tools.ietf.org/html/rfc8536#section-3.2
+    seq:
+      - id: ut_offset
+        -orig-id: utoff
+        type: s4
+        doc: |
+          Number of seconds to be added to UT in order to determine local time.
+          The value MUST NOT be -2**31 and SHOULD be in the range [-89999,
+          93599].
+      - id: is_dst
+        -orig-id: dst
+        type: u1
+        valid:
+          max: 1
+        doc: |
+          Indicates whether local time should be considered Daylight Saving Time
+          (DST). The value MUST be 0 or 1. A value of one (1) indicates that
+          this time type is DST. A value of zero (0) indicates that this time
+          type is standard time.
```

Comment:
A singular attribute beginning with `is_` should be a boolean, not an integer. You should provide a boolean value instance as a supplement to this attribute:

```diff
-      - id: is_dst
+      - id: dst
-        -orig-id: dst
         type: u1
```

```ksy
    instances:
      is_dst:
        value: dst != 0
```

Also I don't really like `valid/max: 1`, it does not express the intent well. Be explicit rather than implicit (i.e. use [`valid/any-of`](https://github.com/kaitai-io/kaitai_struct/issues/435#issue-323332353)):

```ksy
  - id: dst
    type: u1
    valid:
      any-of:
        - 0
        - 1
```

alternatively (if you prefer):

```ksy
  - id: dst
    type: u1
    valid:
      any-of: [0, 1]
```

Reply:
The TZif data also has arrays of u8 that contain [0, 1] values. I'm not sure how to deal with those unless value instances can be used to convert arrays of u8 to arrays of bool.

Reply:
For now, I have renamed it to `dst_flag` and changed the validity to `any-of: [0, 1]`, but have not mapped it to a boolean value instance. I have also renamed the other `is_` attributes of type `u1`.